### ISSUE TITLE
[bug] MacOS titlebar height inclusive while Linux and Windows are exclusive

### ISSUE THREAD

**Issue by ionutrogojan** [tauri-apps/tauri#6333]
### Describe the bug

The height of a window being set in the Tauri config is not respected across all platforms. MacOS considers the title-bar as inclusive of the window height, while Windows and Linux do not.
## tauri.conf.json
```json
"windows": [
  {
    "title": "Icon Frame",
    "theme": "Dark",
    
    "width": 900,
    "maxWidth": 900,
    "minWidth": 900,
    
    "height": 632,
    "maxHeight": 632,
    "minHeight": 632,
    
    "fullscreen": false,
    "resizable": false
  }
```

## Linux window breakdown
![image](https://media.discordapp.net/attachments/1077060020388241439/1077586152275591248/breakdown.png?width=1234&height=657)

## Windows window from dev-tools
![image](https://media.discordapp.net/attachments/1077060020388241439/1077587419681005649/window_size_windows.PNG?width=1227&height=657)

## MacOS window with the inclusive title-bar
![image](https://media.discordapp.net/attachments/1077060020388241439/1077586866087411843/image.png?width=909&height=657)

#

Changing the height inside the `tauri.conf.json` from `632` to `660` to compensate for the `632 - 604` difference does seem to resolve the issue for the moment.

### Reproduction

1. Open the application on MacOS

### Expected behavior

A consistent window height across all 3 supported platforms

### Platform and versions

Environment
  › OS: Fedora 37.0.0 X64
  › Node.js: 19.3.0
  › npm: 9.2.0
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.25.2
  › rustc: 1.67.1
  › cargo: 1.67.1
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
WARNING: no lock files found, defaulting to npm
  › @tauri-apps/cli [NPM]: 1.2.3
  › @tauri-apps/api [NPM]: Not installed!
  › tauri [RUST]: 1.2.4,
  › tauri-build [RUST]: 1.2.1,
  › tao [RUST]: 0.15.8,
  › wry [RUST]: 0.23.4,

App
  › build-type: bundle
  › CSP: default-src 'self'; img-src 'self' asset: https://asset.localhost
  › distDir: ../frontend
  › devPath: ../frontend
package.json not found

App directory structure
  ├─ .git
  ├─ .vscode
  ├─ frontend
  ├─ .github
  ├─ backend
  └─ release-builds

### Additional context

There seems to also be an issue with opening the `dev-tools` on linux. Because the `dev-tools` open as part of the window itself, it seems to force it to resize in order to properly fit inside the window even with `"resizable": false` and an equal `height/maxHeight/minHeight` setting. See dev-tools console output. Expected `632`, got `656`
#
![image](https://media.discordapp.net/attachments/1077060020388241439/1077586424280383520/Screenshot_from_2023-02-21_13-38-43.png?width=1187&height=657)
#
Not really an issue because the dev-tools are disabled inside of the build, just an observation

**Comment by tedsteen** [comment]
I am having the same issue. Is it possible to know how much you should compensate for on the height? I can't seem to find a way to query for the titlebar height.

Same goes for width btw. I would assume it is the scrollbar that is not included on windows.
Works as expected on Mac for both.

**Comment by ionutrogojan** [comment]
@tedsteen I got it working by adding 28px to the height of the window. That was the default window decoration size on MacOs at the time of opening the issue

**Comment by tedsteen** [comment]
> @tedsteen I got it working by adding 28px to the height of the window. That was the default window decoration size on MacOs at the time of opening the issue

Yeah, I am doing a similar thing right now. But the compensation will probably differ between users so it is not optimal.

In my case tho, the MacOS app behaves as expected. The inners size of the window is exactly what I tell it to be, but on Windows (11) it is 35 pixels to high and 12 pixels to wide.
Perhaps compensating for that will work in most cases but I'm sure it won't in all cases..

**Comment by WilfredC64** [comment]
I had the same issue reported some time ago but it was closed because I think it was wrongly interpreted. The issue is still there like described above. Please investigate since this is a real issue.

The bug I reported can be found here: https://github.com/tauri-apps/tauri/issues/4226
